util/set.go

```diff
@@ -40,3 +40,37 @@ func (s Set[T]) Contains(elem T) (exists bool) {
 	_, exists = s[elem]
 	return
 }
+
+// Union constructs a new set, containing all elements from the given sets. nil
+// is never returned
+func Union[T comparable](sets ...Set[T]) Set[T] {
+	union := make(Set[T])
+	for _, set := range sets {
+		for elem := range set {
+			union.Add(elem)
+		}
+	}
+	return union
+}
+
+// Intersection constructs a new set, containing all elements that appear in all
+// given sets. nil is never returned.
+func Intersection[T comparable](sets ...Set[T]) Set[T] {
+	var intersection = make(Set[T])
+	if len(sets) == 0 {
+		return intersection
+	}
+	for elem := range sets[0] {
```

Comment:
Can't find any usages so not sure if this Cn^2 too bad or acceptable

Reply:
It's not used, it just felt weird to write Union without writing Intersection.  I don't think you can do better than `O(m*n)` though (m = number of sets, n = size of sets)

Reply:
I'm pretty surprised they don't have this at least implemented in the exp package - so many homespun hashset implementations around Golang.